packages/legend-query-builder/src/components/QueryBuilderParametersPanel.tsx

```diff
@@ -108,46 +109,47 @@ const VariableExpressionEditor = observer(
     const multilicityOptions: MultiplicityOption[] =
       validParamMultiplicityList.map(buildMultiplicityOption);
     const changeMultiplicity = (val: MultiplicityOption): void => {
-      lambdaParameterState.changeMultiplicity(varState, val.value);
+      setSelectedMultiplicity(val);
     };
 
-    const parameterNameValue = varState.name;
-
+    // name
+    const stateName = varState.name;
+    const [selectedName, setSelectedName] = useState(stateName);
     const [isNameValid, setIsNameValid] = useState<boolean>(true);
 
     const getValidationMessage = (input: string): string | undefined =>
       !input
         ? `Parameter name can't be empty`
-        : allVariableNames.filter((e) => e === input).length >
-          (isCreating ? 0 : 1)
-        ? 'Parameter name already exists'
-        : (isCreating &&
-            queryParametersState.parameterStates.find(
-              (p) => p.parameter.name === input,
-            )) ||
-          (!isCreating &&
-            queryParametersState.parameterStates.filter(
-              (p) => p.parameter.name === input,
-            ).length > 1)
```

Comment:
`allVariableNames` already contains all the names of the variables from `queryParametersState.parameterStates`, so these lines seemed redundant; hence, I removed them. Let me know if there was a reason for having this check here, too.

Reply:
I think it's safe to remove it as the var name got changed immediately before so the check number for create/update is different.